src/gui/libcloudproviders/libcloudproviders.cpp

```diff
@@ -0,0 +1,234 @@
+
+    auto actionGroup = g_simple_action_group_new();
+    const GActionEntry entries[] = {
+        { "settings", actionDispatcher, nullptr, nullptr, nullptr, {}},
+      };
+    g_action_map_add_action_entries(G_ACTION_MAP(actionGroup), entries, G_N_ELEMENTS(entries), this);
+
+    cloud_providers_account_exporter_set_menu_model(exporter, G_MENU_MODEL(menu));
+    cloud_providers_account_exporter_set_action_group(exporter, G_ACTION_GROUP(actionGroup));
+
+    // Currently there's no reason for us to keep these around: no further modifications are done
+    g_clear_object(&menu);
+    g_clear_object(&actionGroup);
+
+    _folderExports[folder] = FolderExport{folder, exporter};
+    updateFolderExport();
+}
+
+void LibCloudProvidersPrivate::unexportFolder(Folder *folder)
+{
+    if (!_folderExports.contains(folder))
+        return;
+    auto folderExporter = _folderExports[folder]._exporter;
+    cloud_providers_provider_exporter_remove_account(_exporter, folderExporter);
+    // the remove_account already calls _unref
+    _folderExports.remove(folder);
+}
+
+void LibCloudProvidersPrivate::updateFolderExport()
+{
+    for (auto folderExport : _folderExports) {
+        if (!folderExport._folder)
+            continue;
+        Folder *folder = folderExport._folder;
+
+        // Update the name, may change if accounts are added/removed
+        QString displayName = folder->shortGuiRemotePathOrAppName();
+        if (AccountManager::instance()->accounts().size() > 1) {
+            displayName = QStringLiteral("%1 (%2)").arg(
+                    displayName, folder->accountState()->account()->displayName());
+        }
+        cloud_providers_account_exporter_set_name(folderExport._exporter, displayName.toUtf8().constData());
+
+        CloudProvidersAccountStatus status = CLOUD_PROVIDERS_ACCOUNT_STATUS_INVALID;
+        auto syncResult = folder->syncResult();
+        switch (syncResult.status()) {
+        case SyncResult::Undefined:
+            status = CLOUD_PROVIDERS_ACCOUNT_STATUS_INVALID;
+            break;
+        case SyncResult::NotYetStarted:
+        case SyncResult::Success:
+        case SyncResult::Problem:
+            status = CLOUD_PROVIDERS_ACCOUNT_STATUS_IDLE;
+            break;
+        case SyncResult::SyncPrepare:
+        case SyncResult::SyncRunning:
+        case SyncResult::SyncAbortRequested:
+            status = CLOUD_PROVIDERS_ACCOUNT_STATUS_SYNCING;
+            break;
+        case SyncResult::Error:
+        case SyncResult::SetupError:
+            status = CLOUD_PROVIDERS_ACCOUNT_STATUS_ERROR;
+            break;
+        case SyncResult::Paused:
+            status = CLOUD_PROVIDERS_ACCOUNT_STATUS_IDLE; // questionable
+            break;
+        }
+
+        // We don't do this in other status reporting, but it would make some sense?
+        //if (!account->_folder->accountState()->isConnected())
+        //    status = CLOUD_PROVIDERS_ACCOUNT_STATUS_ERROR;
```

Comment:
Same as with paused.
If `IDLE` means everything is fine and new files are going to be sync'ed,  then IDLE is not the right choice.
I'd say `ERROR` is better. even for paused, as it means something like:  some account are not synchronized.

Now, that depends on how these status are reported to the user.  I guess the actual information is in the message.